### PR TITLE
Fixed some known issues

### PR DESCRIPTION
**Pull request by JoeyHwong-gk**
- Add docs and code comment #85 
- S3 storage support #85 
- Reduce parameters in lifelong-learning/multi-task-learning #85 
- fix bugs in incremental_learning example - epoch always be 1, inference result not saved
- fix bugs in federated_learning example - some node permanently waiting

### CODE REVIEW

File: lib/sedna/algorithms/multi_task_learning/multi_task_learning.py [hunk 1]
```diff
@@ -68,9 +66,7 @@ def __init__(self,
         self.extractor = None
         self.base_model = estimator
         self.task_groups = None
-        self.task_index_url = Context.get_parameters(
-            "MODEL_URLS", '/tmp/index.pkl'
-        )
+        self.task_index_url = "index.pkl"
         self.min_train_sample = int(Context.get_parameters(
             "MIN_TRAIN_SAMPLE", '10'
         ))
```
Comment (JimmyYang20) [inline lib/sedna/algorithms/multi_task_learning/multi_task_learning.py:72]: put  `MIN_TRAIN_SAMPLE`  to contant.py.
set `10` to a contant

File: lib/sedna/algorithms/multi_task_learning/multi_task_learning.py [hunk 2]
```diff
@@ -68,9 +66,7 @@ def __init__(self,
         self.extractor = None
         self.base_model = estimator
         self.task_groups = None
-        self.task_index_url = Context.get_parameters(
-            "MODEL_URLS", '/tmp/index.pkl'
-        )
+        self.task_index_url = "index.pkl"
```
Comment (JimmyYang20) [inline lib/sedna/algorithms/multi_task_learning/multi_task_learning.py:69]: set a contant

File: lib/sedna/service/server/knowledgeBase/server.py [hunk 3]
```diff
@@ -143,12 +142,10 @@ def update_status(self, data: KBUpdateResult = Body(...)):
...
         task_info["task_groups"] = new_task_group
-        self.latest += 1
-
         _index_path = FileOps.join_path(self.save_dir,
-                                        f"kb_index_{self.latest}.pkl")
-        joblib.dump(task_info, _index_path)
-        res = f"/file/download?files=kb_index_{self.latest}.pkl&name=index.pkl"
+                                        "kb_index.pkl")
+        FileOps.dump(task_info, _index_path)
+        res = "/file/download?files=kb_index.pkl&name=index.pkl"
```
Comment (JimmyYang20) [inline lib/sedna/service/server/knowledgeBase/server.py:148]: set `res` to a contant

File: lib/sedna/service/server/knowledgeBase/server.py [hunk 4]
```diff
@@ -221,15 +213,10 @@ def update(self, task: UploadFile = File(...)):
...
-
         # todo: get from kb
         _index_path = FileOps.join_path(self.save_dir,
-                                        f"kb_index_{self.latest}.pkl")
-        FileOps.upload(name, _index_path)
-        res = f"/file/download?files=kb_index_{self.latest}.pkl&name=index.pkl"
+                                        "kb_index.pkl")
+        _index_path = FileOps.dump(upload_info, _index_path)
+
+        res = "/file/download?files=kb_index.pkl&name=index.pkl"
```
Comment (JimmyYang20) [inline lib/sedna/service/server/knowledgeBase/server.py:221]: same  problem above

**Comment by llhuii** [comment]
Please fix the typo of commit message.

File: examples/lifelong_learning/atcii/README.md [hunk 5]
```diff
@@ -1,8 +1,8 @@
 # Using Lifelong Learning Job in Thermal Comfort Prediction Scenario
 
 This document introduces how to use lifelong learning job in thermal comfort prediction scenario. 
-Using the lifelong learning job, our application can automatically retrains, evaluates, 
-and updates models based on the data generated at the edge.
+Using the lifelong learning job, our application can automatically retrain, evaluate, 
+and update models based on the data generated at the edge.
```
[the remark this answers is not in the document]
Comment (JoeyHwong-gk) [reply examples/lifelong_learning/atcii/README.md:5]: fix

File: lib/sedna/algorithms/multi_task_learning/multi_task_learning.py (hunk 1 above)
Comment (JoeyHwong-gk) [reply lib/sedna/algorithms/multi_task_learning/multi_task_learning.py:72]: fix

File: lib/sedna/service/server/knowledgeBase/server.py (hunk 3 above)
Comment (JoeyHwong-gk) [reply lib/sedna/service/server/knowledgeBase/server.py:148]: fix

File: lib/sedna/service/server/knowledgeBase/server.py (hunk 4 above)
Comment (JoeyHwong-gk) [reply lib/sedna/service/server/knowledgeBase/server.py:221]: fix

**Pull request closed without merging**